File: portal_fejepe/app/models/evento.py
"""
Schemas Pydantic para a aba de Eventos.

Os eventos são calculados a partir dos indicadores já consolidados
das EJs, sem precisar de uma tabela dedicada nesta primeira versão.
"""

from enum import Enum

from pydantic import BaseModel, Field


class EventoMetaTipo(str, Enum):
    """Tipos de meta monitorados no evento."""

    fora_do_zero = "fora_do_zero"
    verde_abril = "verde_abril"
    colab_tracking = "colab_tracking"


class EventoResumo(BaseModel):
    """Resumo de um evento disponível na aba de Eventos."""

    id_evento: str
    nome: str
    descricao: str | None = None
    ativo: bool = True
    ano: int
    mes_referencia: int = Field(..., ge=1, le=12)


class EventoMetaParticipante(BaseModel):
    """EJ contabilizada em uma meta do evento."""

    id_ej: int
    nome: str
    comunidade: str | None = None
    cluster: int | None = None
    status: str | None = None
    faturamento_acumulado: float = 0
    percentual_meta: float | None = None
    projetos_colab_totais: int = 0
    atende_cluster_1_2: bool = False


class EventoMetaResultado(BaseModel):
    """Resultado calculado para uma meta do evento."""

    tipo: EventoMetaTipo
    titulo: str
    descricao: str
    meta_percentual: float
    meta_contagem: int
    resultado_percentual: float
    resultado_contagem: int
    gap_percentual: float
    gap_contagem: int
    submeta_titulo: str | None = None
    submeta_percentual: float | None = None
    submeta_contagem: int | None = None
    subresultado_contagem: int | None = None
    subgap_contagem: int | None = None
    participantes: list[EventoMetaParticipante] = Field(default_factory=list)


class EventoDetalheResponse(BaseModel):
    """Resposta principal da aba de Eventos."""

    evento: EventoResumo
    total_ejs: int
    metas: list[EventoMetaResultado]

File: front_portal_fejepe/src/types/evento.ts
export type EventoMetaTipo = 'faturamento_zero' | 'colab_zero' | 'verde_mes' | 'cluster_tracking';

export interface EventoResumo {
    id_evento: string;
    nome: string;
    descricao: string | null;
    ativo: boolean;
    ano: number;
    mes_referencia: number;
}

export interface EventoMetaParticipante {
    id_ej: number;
    nome: string;
    comunidade: string | null;
    cluster: number | null;
    status: string | null;
    faturamento_acumulado: number;
    faturamento_colab_acumulado: number;
    percentual_meta: number | null;
    projetos_colab_totais: number;
    atende_cluster_1_2: boolean;
    tendencia_cluster?: 'sobe' | 'mantem' | 'desce' | null;
}

export interface EventoMetaResultado {
    tipo: EventoMetaTipo;
    titulo: string;
    descricao: string;
    meta_percentual: number;
    meta_contagem: number;
    resultado_percentual: number;
    resultado_contagem: number;
    gap_percentual: number;
    gap_contagem: number;
    submeta_titulo: string | null;
    submeta_percentual: number | null;
    submeta_contagem: number | null;
    subresultado_contagem: number | null;
    subgap_contagem: number | null;
    participantes: EventoMetaParticipante[];
}

export interface EventoDetalheResponse {
    evento: EventoResumo;
    total_ejs: number;
    metas: EventoMetaResultado[];
}